sgx-types/src/ssa.rs

```diff
@@ -0,0 +1,135 @@
+#![allow(clippy::unreadable_literal)]
+
+#[derive(Copy, Clone, Debug)]
+pub enum Exception {
+    Divider,
+    Debug,
+    Breakpoint,
+    BoundRange,
+    InvalidOpCode,
+    GeneralProtection,
+    PageFault,
+    FloatingPoint,
+    AlignmentCheck,
+    Simd,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum ExitType {
+    Hardware,
+    Software,
+}
+
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ExitInfo(u32);
+
+impl std::fmt::Debug for ExitInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let exc = self.exception();
+        let et = self.exit_type();
+
+        let status = et.and_then(|et| exc.map(|e| (et, e)));
+        write!(f, "ExitInfo({:?})", status)
+    }
+}
+
+impl ExitInfo {
+    pub fn new(et: ExitType, exc: Exception) -> Self {
+        let et = match et {
+            ExitType::Hardware => 0b011 << 8,
+            ExitType::Software => 0b110 << 8,
+        };
+
+        let exc = match exc {
+            Exception::Divider => 0x00,
+            Exception::Debug => 0x01,
+            Exception::Breakpoint => 0x03,
+            Exception::BoundRange => 0x05,
+            Exception::InvalidOpCode => 0x06,
+            Exception::GeneralProtection => 0x0d,
+            Exception::PageFault => 0x0e,
+            Exception::FloatingPoint => 0x10,
+            Exception::AlignmentCheck => 0x11,
+            Exception::Simd => 0x13,
+        };
+
+        ExitInfo(1 << 31 | et | exc)
+    }
+
+    pub fn exit_type(self) -> Option<ExitType> {
+        const MASK: u32 = 1 << 31 | 0b111 << 8;
```

Comment:
Same question about these values.

Reply:
These are in the SGX docs. This section still needs improvement.